### ISSUE TITLE
I am getting value error on training, due to output dimensions even though my output dimensions are same as that of model

### ISSUE THREAD

**Issue by ShantanuShinde** [keras-team/keras#11470]
layers=[Conv2D(8,kernel_size=(4,4),padding='same',input_shape=( 200, 180,3),kernel_initializer="glorot_normal",data_format="channels_first"),
    Activation("relu"),
    MaxPooling2D(pool_size=(8,8),padding='same',data_format='channels_first'),
    Conv2D(16,(2,2),padding='same',kernel_initializer="glorot_normal"),
    Activation("relu"),
    MaxPooling2D(pool_size=(4,4),padding='same',data_format='channels_first'),
    Conv2D(4,(3,3),padding='same',kernel_initializer="glorot_normal"),
    Activation("relu"),
    MaxPooling2D(pool_size=(2,2),padding='same',data_format='channels_first'),
    Flatten(),
    Dense(2,input_shape=(48,)),
    Softmax(axis=-1)]
model = Sequential(layers)

this is my model. I tested for output shape of the softmax, it shows (None,2) which is what i want. I tested my Y_train shape which was (10,2) as expected but when I ran the model it gave error at fit saying softmax_1 expected output of shape (1,) but got (2,). How is it (1,) when clearly the output must be 2?
